### PR TITLE
fix(tron): block self-send + stop overstating network fee (#4863)

### LINKED ISSUE #4863: TRON: self-send not blocked + fee shown despite sufficient bandwidth/energy
Two related problems observed on the TRON send / Verify screen (see attached screenshot: a TRX send where **From and To are the same vault address**, and **Est. network fee = 0.1 TRX** despite the account having available bandwidth and energy).

---

## 1. Self-send is not blocked

The Verify screen allows sending to the **same address** the funds are coming from (From == To). There is no guard preventing a self-transfer, so the user can sign a transaction that just pays fees to move funds to themselves.

**Expected:** block the send (or show an inline error) when the destination equals the sender's own address on the same chain.

**Reference:** the Android client rejects this with a "same address" send error in its send-validation strategy. iOS should apply the same check in the send-form validation before allowing the user to reach Verify / sign.

---

## 2. A fee is shown even when the account has enough bandwidth/energy

For a native TRX transfer where the account has sufficient bandwidth, the real on-chain cost is **0 TRX** (bandwidth is free from the daily quota / staking). The fee service actually computes this correctly — `getBandwidthFeeDiscount(...)` returns `0` for a native transfer with sufficient bandwidth.

However, that correct `0` is then **overridden with a hardcoded default** before it reaches the UI:

`VultisigApp/VultisigApp/Blockchain/Tron/Service/TronService.swift:80-86`
```swift
let calculatedFee = try await calculateTronFee(coin: coin, to: to, memo: memo, isSwap: isSwap)
// For native transfers with a fully bandwidth-discounted fee (calculatedFee == 0),
// fall back to the coin's static `feeDefault` so the UI displays a non-zero "Fees" line.
let finalFee = calculatedFee == 0 ? coin.feeDefault.toBigInt() : calculatedFee
```

So whenever the true fee is `0`, iOS substitutes `coin.feeDefault` (0.1 TRX), which is why the screenshot shows **0.1 TRX** for an account that should transfer for free. This overstates the fee and is misleading.

**Expected:** when the account has sufficient bandwidth, display **0 TRX** (as it truly is) instead of falling back to the static default.

**Reference:** the Android client shows `0 TRX` in this case — the bandwidth-discounted fee is displayed as-is with no non-zero fallback.

---

## Platform

iOS (iPhone/iPad/Mac)

### PR DESCRIPTION
**Pull request by gastonm5**
Fixes two independent defects on the TRON send screen reported in #4863 (both visible in the issue's screenshot: a TRX send where From == To showing an "Est. network fee = 0.1 TRX" despite the account having free bandwidth).

## Defect A — stop overstating the network fee

`TronService.getBlockInfo` computed the correct fee via `calculateTronFee`, then threw it away when it was `0`, substituting the coin's static `feeDefault` (0.1 TRX):

```swift
let finalFee = calculatedFee == 0 ? coin.feeDefault.toBigInt() : calculatedFee
```

But a **successfully computed** `0` occurs **only** for a native TRX transfer with sufficient bandwidth, no memo, and an active destination — the genuinely-free case. TRC20, native-swap, bandwidth-shortfall, memo, and inactive-destination paths all compute a non-zero fee. So the `== 0` fallback fired exactly when the fee really was 0 and overstated it as 0.1 TRX. Android shows `0 TRX`. For native transfers the signing path does not write `feeLimit` (plain `TransferContract`), so `gasFeeEstimation` only drives the UI + the max-amount calc; showing the true `0` is correct.

The fix uses the computed fee directly (`String(calculatedFee)`) and removes the stale comment. To close a gap surfaced in review: `calculateTronFee` swallowed native-fee computation **errors** to `.zero` via `try?`, which is indistinguishable from a genuine free transfer once collapsed. The fallback is therefore **kept, but moved** to the error path only — `(try? await calculateNativeTrxFee(coin:)) ?? coin.feeDefault.toBigInt()` — so a transient account-resource / chain-parameter fetch failure shows a conservative estimate rather than a false `0`, while a genuinely free transfer still shows `0`.

## Defect B — block TRON self-send (Android parity)

The send-form validation pipeline had no guard for destination == sender, so a user could sign a TRON transfer to their own address that just burns fees. Adds a TRON-scoped `validateNotSelfSend()` to `SendDetailsViewModel`, wired into `validateForm()` immediately after address resolution and before the balance checks (so the self-send error wins):

```swift
func validateNotSelfSend() -> Bool {
    guard coin.chain == .tron, !isStakingOperation else { return true }
    guard toAddress == coin.address else { return true }
    setAddressError(message: "sameAddressError")
    return false
}
```

This mirrors android `DefaultSendStrategy.kt` (`if (!isTronStakingOp && srcAddress == dstAddress) throw send_error_same_address`). It is **deliberately TRON-only and excludes staking ops** (freeze/unfreeze are self-directed by design) — matching Android; **not** generalized to other chains (explicit decision). TRON addresses are single-canonical base58check, so an exact string compare matches Android's `srcAddress == dstAddress`.

Adds the new `sameAddressError` localization key to **all 8** locales (en, de, es, hr, it, ko, pt, zh-Hans), reusing android's `send_error_same_address` translations. English (android verbatim): `"The to address must be different from the from address"`.

## Tests

- `TronServiceFeeLimitTests`: flipped the test that pinned the old behavior (was asserting `100_000`) to assert `gasFee == 0` for the sufficient-bandwidth free case; added an insufficient-bandwidth case asserting the real `300_000` sun bandwidth fee; added an account-resource-error case asserting the conservative `100_000` fallback.
- `SendDetailsViewModelValidationTests`: isolated `validateNotSelfSend()` cases (TRON same-address fires `sameAddressError`; staking-op excluded; different address passes; non-TRON same-address passes) plus an end-to-end `validateForm()` test with a real base58 TRON address proving the guard is wired in and fires before the balance check.

## Gate

- Cross-model review loop: each commit codex-reviewed read-only, plus a whole-branch pass — all triaged clean (one major finding on the fee error-path was fixed as described above).
- SwiftLint: clean on all changed files (no new warnings).
- Full suite: `make test` → `** TEST SUCCEEDED **` (3377 tests executed, 1 skipped, 0 failures) on a dedicated en_US simulator (iPhone 17 Pro, iOS 26.5, UDID 802B9C6B-B3EC-4B37-A288-9F0DA03C8024).

## On-device pending

- [ ] TRON send with To == From is blocked with the inline "same address" error.
- [ ] A native TRX send with free bandwidth shows **0 TRX** (not 0.1 TRX).
- [ ] A TRC20 send still shows a real, non-zero fee.
- [ ] A TRON freeze/unfreeze still works (not blocked by the self-send guard).

Closes #4863

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added TRON “self-send” validation to block sending to the sender’s own address, with localized error messaging (except for staking operations).
* **Bug Fixes**
  * Improved native TRX fee estimation to correctly distinguish true zero-fee transfers from bandwidth costs and from fallback fees when fee data can’t be retrieved.
  * UI fee display now follows the computed fee consistently.
* **Tests**
  * Added/updated unit and end-to-end tests covering self-send validation and TRON fee estimation edge cases.

### CODE REVIEW

File: VultisigApp/VultisigApp/Features/Send/ViewModels/SendDetailsViewModel.swift [hunk 1]
```diff
@@ -770,6 +770,17 @@ final class SendDetailsViewModel {
...
     }
 
+    /// TRON self-send guard (parity with android DefaultSendStrategy): a plain
+    /// transfer whose destination is the sender's own address just burns fees.
+    /// Staking ops (freeze/unfreeze) are self-directed by design, so they're
+    /// excluded via the existing `isStakingOperation` flag.
+    func validateNotSelfSend() -> Bool {
+        guard coin.chain == .tron, !isStakingOperation else { return true }
+        guard toAddress == coin.address else { return true }
+        setAddressError(message: "sameAddressError")
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Features/Send/ViewModels/SendDetailsViewModel.swift:783]: Fixed in be4d1a1bb — the guard now compares `toAddress == fromAddress` (the sender `makeTransaction()` actually signs with) instead of `coin.address`, so it tracks the real sender rather than an assumed-equal default.

Note this is latent today: every `SendDetailsSeed` sets `fromAddress = coin.address` (lines 54/90/348), so no current flow diverges — but keying on `fromAddress` removes the reliance on that unenforced invariant. Added a test pinning the decoupled case (blocks when `to == fromAddress != coin.address`; allows `to == coin.address != fromAddress`). Full suite green (3378 tests, 0 failures).

**Review by johnnyluo — APPROVED** [review]
LGTM
